### PR TITLE
feat(asset-index): add check for decimals of asset metadata

### LINKED ISSUE #363: TOB-PINT-3 - Data validation
Missing decimal validation in the set_metadata function.
Description
The function set_metadata in figure 3.1 doesn't validate the decimals variable lower and upper bound. This allows the governance committee to set the decimals of a given asset to 0 or to a very high value, which are undesired.

The set_metadata function (PINT/pallets/asset-index/src/lib.rs#L472-L493

### PR DESCRIPTION
**Pull request by clearloop**
## Changes

- introduce `MaxDecimals`

## Tests

```
CI
```

## Issues

- Closes #363

### CODE REVIEW

File: runtime/common/src/constants.rs [hunk 1]
```diff
@@ -118,6 +118,7 @@ parameter_types! {
 	pub const TreasuryPalletId: PalletId = PalletId(*b"Treasury");
 	pub const LockupPeriod: BlockNumber = 10;
 	pub const MaxCandidates: u32 = 200;
+	pub const MaxDecimals: u8 = 12;
```
Comment (mattsse) [inline runtime/common/src/constants.rs:121]: we should bump this to 18

File: pallets/asset-index/src/lib.rs [hunk 2]
```diff
@@ -104,6 +104,10 @@ pub mod pallet {
 		#[pallet::constant]
 		type MaxActiveDeposits: Get<u32>;
 
+		/// Restricts how many deposits can be active
+		#[pallet::constant]
+		type MaxDecimals: Get<u8>;
```
Comment (mattsse) [inline pallets/asset-index/src/lib.rs:109]: This comment is not correct.

File: pallets/asset-index/src/lib.rs [hunk 3]
```diff
@@ -325,6 +329,8 @@ pub mod pallet {
 		/// Thrown if deposits reach limit
 		TooManyDeposits,
 		/// Thrown when the given DepositRange is invalid
+		InvalidDecimals,
+		/// Thrown when the given decimals is zero or too high
```
Comment (mattsse) [inline pallets/asset-index/src/lib.rs:333]: comments switched

**Review by mattsse — CHANGES_REQUESTED** [review]
I don't think this is a critical feature since we don't actually use the metadata.
But it's nice to have it.

**Comment by clearloop** [comment]
> I don't think this is a critical feature since we don't actually use the metadata.

mb can leave it to the client who builds Apps around PINT

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**